app/validators/measure_validator.rb

```diff
@@ -151,6 +151,52 @@ class MeasureValidator < TradeTariffBackend::Validator
        record.export_refund_nomenclature.number_indents <= (record.measure_type.measure_explosion_level))))
   end
 
+  #validation :ME104,
+    # WIP
+    #%Q{ "The justification regulation must be either:
+          #- the measure’s measure-generating regulation, or
+          #- a measure-generating regulation, valid on the day after the measure’s (explicit) end date.
+        #If the measure’s measure-generating regulation is ‘approved’, then so must be the justification regulation" } do |record|
+
+    #valid = true
+
+    # # Keeping this code verbose for now (As I am still understanding this code)
+    #if record.justification_regulation_id.present? && record.measure_generating_regulation_id.present?
+      #valid = record.justification_regulation_id == record.measure_generating_regulation_id
+
+      #if record.generating_regulation.present? && record.generating_regulation.validity_end_date.present?
+        #valid = record.generating_regulation.validity_end_date > record.validity_end_date
+      #else
+        #valid = record.validity_end_date.present?
+      #end
+    #end
+
+    #valid
+  #end
+
+  validation :ME112, "If the additional code type has as application 'Export Refund for Processed Agricultural Goods' then the measure does not require a goods code." do |record|
+    valid = true
+
+    if record.additional_code_type.present?
+      valid = record.additional_code_type.description == "Export refund for processed agricultural goods" &&
```

Comment:
I cannot see this in the DB, I think this one needs a bit more work as at the moment it will not do anything

Reply:
There is the additional_code_type

9 | EN | Export Refunds |   | 7 | C |   |   |   |  
-- | -- | -- | -- | -- | -- | -- | -- | -- | --

Reply:
In my database, it shows these values:
```ruby
AdditionalCodeType.all.collect(&:description)
=> ["Anti-dumping/countervailing",
 "Agricultural Tables (Meursing)",
 "Prohibition/Restriction/Surveillance",
 "Restrictions",
 "Agricultural Tables (non-Meursing)",
 "Tariff preference",
 "Export Refunds",
 "Anti-dumping/countervailing",
 "Dual Use",
 "Refund for basic products",
 "Anti-dumping/countervailing",
 "Anti-dumping/countervailing",
 "Reference prices fishery products"]
```
and I thought we might have more descriptions in the production database which might have this value **Export refund for processed agricultural goods**. I am going to use this info **Export Refunds** for now for these rules ME112 and ME113 but not sure how It will get matched with to this extra info *processed agricultural goods*.

Reply:
the description "Export refund for processed agricultural goods" does not exist, so currently the rule will never fire. 

Use the record I found, and I will confirm if there is another way to determine processed agricultural goods, but I think Export Refunds, are only used for these.

Reply:
This is done. Updated it.